Northstar.CustomServers/mod/scripts/vscripts/_loadouts_mp.gnut

```diff
@@ -303,7 +323,7 @@ void function TryGivePilotLoadoutForGracePeriod( entity player )
 		else
 			respawnTimeReal = expect float( player.s.respawnTime )
 		
-		if ( ( ( Time() - respawnTimeReal <= CLASS_CHANGE_GRACE_PERIOD || GetGameState() < eGameState.Playing ) && file.loadoutGracePeriodEnabled ) || player.p.usingLoadoutCrate )
+		if ( ( ( Time() - respawnTimeReal <= CLASS_CHANGE_GRACE_PERIOD || GetGameState() < eGameState.Playing ) && file.loadoutGracePeriodEnabled ) || player.p.usingLoadoutCrate || GameModeRulesShouldGivePilotLoadout( player ) )
```

Comment:
Do you intend for the player loadout checks to only prevent a player from selecting their loadout outside of the grace period? If so, then this functions as desired, but if not you may want to do `if ( ( ( Time() - respawnTimeReal <= CLASS_CHANGE_GRACE_PERIOD || GetGameState() < eGameState.Playing ) && file.loadoutGracePeriodEnabled && GameModeRulesShouldGivePilotLoadout( player ) ) || player.p.usingLoadoutCrate )` instead. Assuming that `player.p.usingLoadoutCrate` should not be overriden, I don't know what that is used for.

Reply:
> Do you intend for the player loadout checks to only prevent a player from selecting their loadout outside of the grace period? If so, then this functions as desired, but if not you may want to do `if ( ( ( Time() - respawnTimeReal <= CLASS_CHANGE_GRACE_PERIOD || GetGameState() < eGameState.Playing ) && file.loadoutGracePeriodEnabled && GameModeRulesShouldGivePilotLoadout( player ) ) || player.p.usingLoadoutCrate )` instead. Assuming that `player.p.usingLoadoutCrate` should not be overriden, I don't know what that is used for.
> 
> If this is supposed to only allow loadout swaps as desired and not deny them, then it looks fine to me.

yeah,it's only allow loadout swaps as desired.I don't want to break too much code